File: backend/src/main/java/EazyTech/EazyHire/services/HiringRoundService.java
package EazyTech.EazyHire.services;

import EazyTech.EazyHire.models.dtos.HiringRoundDTO;
import EazyTech.EazyHire.models.dtos.HiringRoundRequestDTO;
import java.util.List;

public interface HiringRoundService {
    List<HiringRoundDTO> getRounds(Long jobId, Long companyId, Long userId);
    HiringRoundDTO createRound(Long jobId, Long companyId, Long userId, HiringRoundRequestDTO request);
    HiringRoundDTO updateRound(Long jobId, Long roundId, Long companyId, Long userId, HiringRoundRequestDTO request);
    void deleteRound(Long jobId, Long roundId, Long companyId, Long userId);
    List<HiringRoundDTO> reorderRounds(Long jobId, Long companyId, Long userId, List<Long> orderedIds);
}

File: frontend/src/services/hiring-round.service.ts
import { api } from './api';
import type { BaseResponse } from '@/types/api.types';
import type { HiringRound, HiringRoundRequest } from '@/types/hiring-round.types';

export const hiringRoundService = {
  getRounds: async (jobId: string) => (await api.get<BaseResponse<HiringRound[]>>(`/jobs/${jobId}/rounds`)).data.data,
  createRound: async (jobId: string, data: HiringRoundRequest) => (await api.post<BaseResponse<HiringRound>>(`/jobs/${jobId}/rounds`, data)).data.data,
  updateRound: async (jobId: string, roundId: number, data: HiringRoundRequest) => (await api.put<BaseResponse<HiringRound>>(`/jobs/${jobId}/rounds/${roundId}`, data)).data.data,
  deleteRound: async (jobId: string, roundId: number) => (await api.delete<BaseResponse>(`/jobs/${jobId}/rounds/${roundId}`)).data,
  reorderRounds: async (jobId: string, orderedIds: number[]) => (await api.put<BaseResponse<HiringRound[]>>(`/jobs/${jobId}/rounds/reorder`, { orderedIds })).data.data,
};
